pkg/app/app.go

```diff
@@ -172,21 +172,15 @@ func CreateApp() App {
 	// Initialize Runtime
 	agnosticruntime.NewRuntime(app.log.Logger)
 
-	app.log.Debug("Connecting Dao")
-	app.dao, err = dao.NewDao(app.config.Dao, app.log.Logger)
-
 	k8scli, err := clients.Kubernetes(app.log.Logger)
```

Comment:
I'm curious, based on the answer to my previous question, if this getting of a Kubernetes client, getting a k8s rest client, and checking the k8s version is something we should push down into the `NewRuntime` function. That way these checks could be collocated and maybe even simplify the error handling on app startup.

Reply:
This validation could be behind the runtime pkg.  Theoretically, the runtime can be something other than k8s or openshift so we shouldn't assume.

Reply:
isn't that, even more, reason to move the initialization of the client into just the runtime?